src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java

```diff
@@ -33,8 +32,41 @@ public void setTextArea(String text) {
         textArea.setText(text);
     }
 
-    public void appendTextArea(String text) {
-        System.out.println(text);
-        textArea.append(text + "\n");
+    /**
+     * append text in narrative panel with customized foreground color
+     * enable and disable capability to edit before and after append
+     * @param text
+     * @param color
+     */
+    public void appendTextArea(String text, Color color)
+    {
+        this.enableNarrative();
+        this.appendToPane(this.textArea, text, color);
+        this.disableNarrative();
+    }
+
+    public void disableNarrative(){
+        textArea.setEditable(false);
```

Comment:
Can you explain again what is the purpose of using setEditable here?

Reply:
when i setEditable(false), it is still isEnabled() = true, but it won't allow user to edit the content inside textArea

Reply:
if it is not editable, it blocks all color changes